pkg/test/expectations/expectations.go

```diff
@@ -123,31 +124,22 @@ func ExpectDeleted(ctx context.Context, c client.Client, objects ...client.Objec
 }
 
 func ExpectCleanedUp(ctx context.Context, c client.Client) {
-	pdbs := v1beta1.PodDisruptionBudgetList{}
-	Expect(c.List(ctx, &pdbs)).To(Succeed())
-	for i := range pdbs.Items {
-		ExpectDeleted(ctx, c, &pdbs.Items[i])
-	}
-	pods := v1.PodList{}
-	Expect(c.List(ctx, &pods)).To(Succeed())
-	for i := range pods.Items {
-		ExpectDeleted(ctx, c, &pods.Items[i])
-	}
-	nodes := v1.NodeList{}
-	Expect(c.List(ctx, &nodes)).To(Succeed())
-	for i := range nodes.Items {
-		ExpectDeleted(ctx, c, &nodes.Items[i])
-	}
-	daemonsets := appsv1.DaemonSetList{}
-	Expect(c.List(ctx, &daemonsets)).To(Succeed())
-	for i := range daemonsets.Items {
-		ExpectDeleted(ctx, c, &daemonsets.Items[i])
-	}
-	provisioners := v1alpha5.ProvisionerList{}
-	Expect(c.List(ctx, &provisioners)).To(Succeed())
-	for i := range provisioners.Items {
-		ExpectDeleted(ctx, c, &provisioners.Items[i])
+	wg := sync.WaitGroup{}
+	for _, object := range []client.Object{
+		&v1.Pod{},
+		&v1.Node{},
+		&appsv1.DaemonSet{},
+		&v1beta1.PodDisruptionBudget{},
+		&v1.PersistentVolumeClaim{},
+		&v1alpha5.Provisioner{},
+	} {
+		wg.Add(1)
+		go func(object client.Object) {
+			Expect(c.DeleteAllOf(ctx, object, client.InNamespace("default"))).ToNot(HaveOccurred())
```

Comment:
Why are we restricting just to the default namespace here? Shouldn't we be deleting every namespace (regardless if we have tests with non-default namespaces or not) when we clean up?

Reply:
The DeleteAllOf api call requires all namespaces. I suppose you're right that I should describe all namespaces and include them.